Coding/fw/src/CANAdapter.h

```diff
@@ -0,0 +1,259 @@
+/***************************************************************************************************
+  (c) NewTec GmbH 2020   -   www.newtec.de
+  $URL: https://github.com/NewTec-GmbH/esp32-can-iot $
+***************************************************************************************************/
+/**
+@addtogroup Adapters
+@{
+@file       CANAdapter.h
+
+ESP32SJA1000 Adapter for Lawicel Protocol
+
+* @}
+***************************************************************************************************/
+#ifndef CAN_ADAPTER_H
+#define CAN_ADAPTER_H
+
+/* INCLUDES ***************************************************************************************/
+#include <CANInterface.h>
+#include <CAN.h>
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* FORWARD DECLARATIONS ***************************************************************************/
+
+/**
+ *  ESP-32 Adapter as implementation of CANInterface for the Lawicel Protocol.
+ */
+class CANAdapter : public CANInterface
+{
+public:
+    /* CONSTANTS ******************************************************************************/
+
+    /* TYPES **********************************************************************************/
+
+    /**
+    * Default constructor creates instance of the class using default values.
+    * Uses CANInterface constructor as its the implementation of an Interface
+    */
+    CANAdapter() : CANInterface(), m_baudRate(0), m_currentState(CLOSED)
+    {
+    }
+
+    /**
+    * Default destructor deletes instance of the class.
+    */
+    ~CANAdapter()
+    {
+    }
+
+    /** 
+    * Configures and starts the CAN Controller to use the user values.
+    * @return isError = 0 for OK, 1 for Error 
+    */
+    uint8_t begin()
+    {
+        uint8_t isError = 0;
+        if (CAN.begin(500E3) == 0)      /**< Starts CAN channel with 500kbps Baudrate */
+        {
+            isError = 1;
+        }
+        else
+        {
+            CAN.sleep();
+        }
+
+        return isError;
+    }
+
+    /** 
+    * Stops the Controller Module without destroying the instance.
+    * @return isError = 0 for OK, 1 for Error 
+    */
+    uint8_t end()
+    {
+        CAN.end();
```

Comment:
YOu should propagate errors from CAN

like return (CAN.end() == OK) ? 0 : 1

bthw: CAN is not spellled according to coding style. should be canIf or something similar